QModManager/API/QModAPI.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace QModManager.API
+{
+    public static partial class QModAPI
+    {
+        internal static List<Assembly> ErroredMods = new List<Assembly>();
+        public static void MarkAsErrored(Assembly modAssembly = null)
+        {
+            if (ErroredMods.Contains(modAssembly)) return;
+
+            modAssembly = modAssembly ?? Assembly.GetCallingAssembly();
+            ErroredMods.Add(modAssembly);
+        }
+
+        public static ReadOnlyCollection<QMod> GetAllMods(bool includeUnloaded = false, bool includeErrored = false)
+        {
+            if (includeErrored)
+                return Patcher.foundMods.AsReadOnly();
+            else if (includeUnloaded)
+                return Patcher.sortedMods.AsReadOnly();
+            else
+                return Patcher.loadedMods.AsReadOnly();
+        }
+
+        public static QMod GetMod(Assembly modAssembly = null, bool includeUnloaded = false, bool includeErrored = false)
```

Comment:
Recommendation:  
Add a separate method for `GetMyMod()` that returns the calling assembly's QMod instance.  
After that, the `GetMod` can always require a `modAssembly` parameter.